packages/venia-ui/lib/components/CheckoutPage/ShippingMethod/shippingMethod.js

```diff
@@ -1,46 +1,114 @@
+
+import { mergeClasses } from '../../../classify';
+
+import Done from './done';
+import EditForm, { modes as editFormModes } from './editForm';
+import UpdateModal from './updateModal';
 
 import defaultClasses from './shippingMethod.css';
 
+import {
+    GET_SHIPPING_METHODS,
+    GET_SELECTED_SHIPPING_METHOD,
+    SET_SHIPPING_METHOD
+} from './shippingMethod.gql';
+
 const ShippingMethod = props => {
-    const { onSave } = props;
-
-    // TODO: Replace "doneEditing" with a query for existing data.
-    const [doneEditing, setDoneEditing] = useState(false);
-    const handleClick = useCallback(() => {
-        setDoneEditing(true);
-        onSave();
-    }, [onSave]);
-
-    const className = doneEditing
-        ? defaultClasses.container
-        : defaultClasses.container_edit_mode;
-
-    /**
-     * TODO
-     *
-     * Change this to reflect diff UI in diff mode.
-     */
-    const shippingMethod = doneEditing ? (
-        <div>In Read Only Mode</div>
-    ) : (
-        <div>In Edit Mode</div>
-    );
-    return (
-        <div className={className}>
-            <div>Shipping Method Will be handled in PWA-179</div>
-            <div className={defaultClasses.text_content}>{shippingMethod}</div>
-            {!doneEditing ? (
-                <div className={defaultClasses.proceed_button_container}>
-                    <Button onClick={handleClick} priority="normal">
-                        {'Continue to Payment Information'}
-                    </Button>
+    const { onSave, pageIsUpdating, setPageIsUpdating } = props;
+
+    const talonProps = useShippingMethod({
+        mutations: {
+            setShippingMethod: SET_SHIPPING_METHOD
+        },
+        onSave,
+        queries: {
+            getShippingMethods: GET_SHIPPING_METHODS,
+            getSelectedShippingMethod: GET_SELECTED_SHIPPING_METHOD
+        },
+        setPageIsUpdating
+    });
+
+    const {
+        displayState,
+        handleCancelUpdate,
+        handleSubmit,
+        isLoadingShippingMethods,
+        isLoadingSelectedShippingMethod,
+        isUpdateMode,
+        selectedShippingMethod,
+        shippingMethods,
+        showUpdateMode
+    } = talonProps;
+
+    const classes = mergeClasses(defaultClasses, props.classes);
+
+    let contents;
+    switch (displayState) {
+        case displayStates.EDITING:
+            {
+                contents = (
+                    <div className={classes.root}>
+                        <div className={classes.editingBody}>
+                            <h3 className={classes.editingHeading}>
+                                {'Shipping Method'}
+                            </h3>
+                            <EditForm
+                                handleSubmit={handleSubmit}
+                                isLoading={
+                                    isLoadingSelectedShippingMethod ||
+                                    isLoadingShippingMethods
```

Comment:
Try creating another variable for this one.

Reply:
Resolved by a previous commit.